src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -465,8 +465,28 @@ else if (groupIdx > lastGroup) {
      * @return true if imports group are separated internally.
      */
     private boolean checkSeparatorInGroup(int groupIdx, boolean isStatic, int line) {
```

Comment:
please try to inline this method to doVisitToken, it is better to keep madness  of `!beforeFirstImport && ...... && line - lastImportLine > 1` in same method.
is some complexity rule is not allowing it, please rename it to "isSeparatorInGroup".

Reply:
RedundantModifier exception is caught when I add `final` modifier to method.

Reply:
What for you put final modifier to method ?  It is not required.

Please remove this method and put I'd content in visitToken method

Reply:
Inlining this method to doVisitToken directly,value of CyclomaticComplexity is 11.

Reply:
ok, it is better to follow CyclomaticComplexity.
please continue with method renaming.

Reply:
done